### PR TITLE
fix: gifs firefox crash

### PR DESCRIPTION
**Pull request by pravusjif**
Added chrome check to use the unity plugin GIF processing on browsers with no 2D context in OffsetCanvas (like firefox)

### CODE REVIEW

File: kernel/packages/unity-interface/BrowserInterface.ts [hunk 1]
```diff
@@ -343,6 +344,13 @@ export class BrowserInterface {
   }
 
   async RequestGIFProcessor(data: { imageSource: string; id: string; isWebGL1: boolean }) {
+    const isChrome = !!window.chrome && (!!window.chrome.webstore || !!window.chrome.runtime)
```
Comment (BrianAmadori) [inline kernel/packages/unity-interface/BrowserInterface.ts]: It shouldn't be better to use this check? So the gif still works on non-chrome browsers with offscreen canvas down the road? (and if firefox add support it'd fix itself)

```
if (typeof OffscreenCanvas !== "undefined") {     
    isSupported = true; 
} else {     
    isSupported = false; 
}
```

Comment (pravusjif) [reply kernel/packages/unity-interface/BrowserInterface.ts]: good idea, thx!

File: kernel/packages/unity-interface/BrowserInterface.ts [hunk 2]
```diff
@@ -343,6 +344,14 @@ export class BrowserInterface {
   }
 
   async RequestGIFProcessor(data: { imageSource: string; id: string; isWebGL1: boolean }) {
+    // tslint:disable-next-line
+    const isSupported = typeof OffscreenCanvas !== "undefined"
```
Comment (pablitar) [inline kernel/packages/unity-interface/BrowserInterface.ts]: Have you tested this in Firefox? Keep in mind that `OffscreenCanvas` seems to be undefined only in worker context, and not in the main thread.

Also, we should test if the OffscreenCanvas has `2d context` support.

Comment (pravusjif) [reply kernel/packages/unity-interface/BrowserInterface.ts]: I'll add the `typeof OffscreenCanvasRenderingContext2D === "function"` check, thx for pointing it out!

**Review by BrianAmadori — APPROVED** [review]

**Review by sandrade-dcl — APPROVED** [review]
Code reviewed and fix tested on Firefox: I have loaded the Superrare scene where there are a lot of gif's and the navigator have not crashed. The GIF's have been loaded slower than in Chrome version, just as we expected.

**Review by pablitar — APPROVED** [review]
Great work!! :D

**Pull request merged**